app/models/zone.rb

```diff
@@ -78,6 +107,11 @@ def self.default_zone
     in_my_region.find_by(:name => "default")
   end
 
+  # Zone for paused providers (no servers in it), not visible by default
+  cache_with_timeout(:maintenance_zone) do
+    MiqRegion.my_region&.maintenance_zone || create_maintenance_zone
```

Comment:
I don't think this should create here...

Reply:
This is your own request from 27th September, @bdunne :)

Reply:
haha, I can't even find the comment from 27th September... thanks rails and the "14 days ago" convenience junk..

Either way, people won't expect a cached query method to create something as a side effect.

Reply:
so, let's look for some final solution:
`cache_with_timeout` - yes
`create_maintenance_zone` - only in Zone.seed
Could it be?

Reply:
Yes!  Call `create_maintenance_zone` or the body of that method from `Zone.seed` and leave the cached method as a simple query method that will AND should return a nil if the maintenance zone doesn't exist.

Reply:
:+1: